Module.md

```diff
@@ -0,0 +1,189 @@
+The Azure Cosmos DB is Microsoft’s NOSQL database in Azure technology stack. It is called a globally distributed 
+multi-model database which is used for managing data across the world. Key purposes of the Azure CosmosDB is to achieve 
+low latency and high availability while maintaining a flexible scalability. 
+The Ballerina Cosmos DB connector allows you to connect to a Azure Cosmos DB resource from Ballerina and perform various 
+operations such as `find`, `create`, `read`, `update`, and `delete` operations of `Databases`, `Containers`,
+`User Defined Functions`, `Tiggers`, `Stored Procedures`, `Users`, `Permissions` and `Offers`. 
+
+## Compatibility
+
+|                           |    Version                  |
+|:-------------------------:|:---------------------------:|
+| Ballerina Language        | Swan-Lake-Preview8          |
+| Cosmos DB API Version     | 2018-12-31                  |
+
+## CosmosDB Clients
+
+There are two clients provided by Ballerina to interact with CosmosDB.
+
+1. **cosmosdb:DataPlaneClient** - This connects to the running CosmosDB databases and containers to execute data-plane 
+operations 
+
+   ```ballerina
+    cosmosdb:Configuration configuration = {
+        baseUrl : <BASE_URL>,
+        masterOrResourceToken : <MASTER_OR_RESOURCE_TOKEN>,
+    };
+    cosmosdb:DataPlaneClient coreClient = new(configuration);
+   ```
+2. **cosmosdb:ManagementClient** - This connects to the running CosmosDB databases and containers to execute 
+management-plane operations 
+
+   ```ballerina
+    cosmosdb:Configuration configuration = {
+        baseUrl : <BASE_URL>,
+        masterOrResourceToken : <MASTER_OR_RESOURCE_TOKEN>,
+    };
+    cosmosdb:ManagementClient managementClient = new(configuration);
+   ```
+
+## Samples 
+### Creating a database
+For creating a database in Azure we have to provide a unique database ID which does not already exist in the specific 
+cosmos DB account. This operation will return a record of type Database. This will contain the success as true if the 
+operation is successful.
+
+```ballerina
+import ballerina/log;
+import ballerinax/azure_cosmosdb as cosmosdb;
+
+public function main() {
+    cosmosdb:Configuration configuration = {
+        baseUrl : "https://cosmosconnector.documents.azure.com:443",
+        masterOrResourceToken : "mytokenABCD==",
+    };
+    cosmosdb:ManagementClient managementClient = new(configuration);
+
+    cosmosdb:Database databaseResult = checkpanic managementClient->createDatabase(<DATABASE_ID>);
+}
+```
+
+### Creating a Container
+A container can be created inside an existing database in the cosmos DB account. As the REST api version which is used 
+in this implementation of the connector strictly supports the partition key, it is a necessity to provide the 
+partition key in the creation of a container. 
+
+```ballerina
+import ballerina/log;
+import ballerinax/azure_cosmosdb as cosmosdb;
```

Comment:
Why the module name is `azure_cosmosdb`? Shouldn't it be `azure.cosmosdb`?

Reply:
For now the connector is in SLP8. So, it doesn't support adding dots in the module name. Once the connector migrate to slalpha1 this name will be changed as `azure.cosmosdb`

Reply:
ack.